### PR TITLE
495-add-metrics-to-briefs-user-and-admin-frontends

### PR DESCRIPTION
**Pull request by benvand**
* Add dependencies
* Add metrics
* Add tests

https://trello.com/c/Fqq8RCEb/495-add-metrics-to-briefs-user-and-admin-frontends

### CODE REVIEW

File: app/metrics.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+from flask import Blueprint
+from flask.signals import got_request_exception, request_finished
+
+from gds_metrics import GDSMetrics
+
+
+metrics = Blueprint('metrics', __name__)
+
+
+class DMGDSMetrics(GDSMetrics):
```
Comment (katstevens) [inline app/metrics.py]: Didn't we move this to utils? Am I getting confused?

Comment (benvand) [reply app/metrics.py]: Good shout, copied the wrong file

File: tests/helpers.py [hunk 2]
```diff
@@ -19,12 +19,16 @@ def setup_method(self, method):
         # as the DataAPIClient instance has already been created; nor can we temporarily replace app.data_api_client
         # with a mock, because then the shared instance won't have been configured (done in create_app). Instead,
         # just mock the one function that would make an API call in this case.
+        self.app_env_var_mock = patch.dict('gds_metrics.os.environ', {'PROMETHEUS_METRICS_PATH': '/_metrics'})
```
Comment (lfdebrux) [inline tests/helpers.py]: I'm confused, should this be `/user/_metrics` as in the tests below?

Comment (benvand) [reply tests/helpers.py]: No, we register the metrics blueprint prefixed at `'/user'` in the app setup here:
https://github.com/alphagov/digitalmarketplace-user-frontend/pull/119/files#diff-1f9f743421417fb2794c88447083a031R33

The `gds_metrics` library defines the url suffix using:
`gds_metrics.os.environ['PROMETHEUS_METRICS_PATH']`
here:
https://github.com/alphagov/gds_metrics_python/blob/master/gds_metrics/__init__.py#L30

The tests set up the app the same as it is in prod, we initialise the metrics which grabs `_metrics` from the `PROMETHEUS_METRICS_PATH` env var, then we prefix the metrics blueprint with `/user`

Comment (lfdebrux) [reply tests/helpers.py]: Okay gotcha

**Pull request merged**